datahub-web-react/src/app/entity/shared/containers/profile/EntityProfile.tsx

```diff
@@ -245,7 +245,13 @@ export const EntityProfile = <T, U>({
             },
         })) || [];
 
-    const routedTab = useRoutedTab([...tabsWithDefaults, ...autoRenderTabs]);
+    const enabledAndVisibleTabs = [...tabsWithDefaults, ...autoRenderTabs].filter(
```

Comment:
Okay one more small question. 

Why not just pass "visibleTabs" into the EntityTabs component?

Then the EntityTabs component will _always_ only be aware of visible tabs. This makes sense since other tabs will not be involved in rendering. Then, it can properly route the routedTab (if it is not undefined AND it is enabled -- we can keep that check in the EntityTabs component)

In short, the EntityProfile will filter for tabs which are _visibile_ at all, while the child component, EntityTabs, will filter for tabs which are _enabled_.

It keeps the responsibilities fairly clearly separated, to avoid having _both_ places check _both_ visible and enabled statuses.

Reply:
This is something I was not really sure about it, but now only visibleTabs are passed to EntityTabs.